spec/unit/action/deploy/module_spec.rb

```diff
@@ -182,7 +182,7 @@
     subject { described_class.new({ config: '/some/nonexistent/path' }, ['mod1', 'mod2'], {}) }
 
     let(:cache) { instance_double("R10K::Git::Cache", 'sanitized_dirname' => 'foo', 'cached?' => true, 'sync' => true) }
-    let(:repo) { instance_double("R10K::Git::StatefulRepository", cache: cache, resolve: 'main') }
+    let(:repo) { instance_double("R10K::Git::StatefulRepository", cache: cache, resolve: 'main', tracked_paths: []) }
```

Comment:
When we call `determine_purge_exclusions` we ask the environment for its `desired_contents`. A Git backed environment will call `tracked_paths` on it's backing repo (here: https://github.com/puppetlabs/r10k/blob/main/lib/r10k/environment/git.rb#L90)

Reply:
ah, so previously we were getting away with not including `tracked_paths` here because this test used to not reach a point where we needed the `desired_contents`, yes? this is just some stubbing weirdness and a real `R10K::Git::StatefulRepository` would have always had `tracked_paths`, even before this PR?

Reply:
Yeah.